src/System.Net.Security/tests/FunctionalTests/SslStreamStreamToStreamTest.cs

```diff
@@ -121,6 +121,30 @@ public void SslStream_StreamToStream_Successive_ClientWrite_Async_Success()
             }
         }
 
+        [Fact]
+        public void SslStream_StreamToStream_Write_ReadByte_Success()
+        {
+            MockNetwork network = new MockNetwork();
+
+            using (var clientStream = new FakeNetworkStream(false, network))
+            using (var serverStream = new FakeNetworkStream(true, network))
+            using (var clientSslStream = new SslStream(clientStream, false, AllowAnyServerCertificate))
+            using (var serverSslStream = new SslStream(serverStream))
+            {
+                bool result = DoHandshake(clientSslStream, serverSslStream);
+                Assert.True(result, "Handshake completed.");
+
+                for (int i = 0; i < 3; i++)
+                {
+                    clientSslStream.Write(_sampleMsg);
```

Comment:
It would be interesting to write a large enough block that would exceed the internal buffer and force the 1-byte `Read` within the new `ReadByte` override.
(https://github.com/stephentoub/corefx/blob/sslreadbyte/src/System.Net.Security/src/System/Net/SecureProtocols/SslStreamInternal.cs#L22)